packages/babel-helper-create-class-features-plugin/src/fields.ts

```diff
@@ -188,9 +189,26 @@ const privateNameVisitor = privateNameVisitorFactory<
   },
 });
 
+// rename all bindings that shadows innerBinding
+function unshadow(
+  name: string,
+  scope: Scope,
+  innerBinding: t.Identifier | undefined,
+) {
+  const binding = scope.getBinding(name);
+  if (innerBinding && binding && innerBinding !== binding.identifier) {
```

Comment:
The `if` branch is an inlined version of `Scope#bindingIdentifierEquals`, guess we can just put
```js
while (!scope.bindingIdentifierEquals(name, innerBinding)) {
      scope.rename(name);
      scope = scope.parent;
    }
```
here?

Reply:
Sounds great! And all tests passed.

But I got a question: there are some cases that `scope.bindingIdentifier(name) === undefined` and `innerBinding !== undefined` which means `scope.bindingIdentifierEquals(name, innerBinding)` will always returns `false`. This will cause the while loop runs all the way alone to the top level scope and renaming all the variables.

The renaming and looping here are useless, and maybe cause performance problem.

So maybe we change it to:

```ts
  while (
    scope?.hasBinding(name) &&
    !scope.bindingIdentifierEquals(name, innerBinding)
  ) {
    scope.rename(name);
    scope = scope.parent;
  }
```

Reply:
> there are some cases that scope.bindingIdentifier(name) === undefined and innerBinding !== undefined

If the `scope` is the parent scope of the class where `innerBinding` is defined, then `scope.bindingIdentifier(classRef.name)` is surely `undefined`, unless defined otherwise. We could exit the loop after scope becomes the class scope.